### PR TITLE
enh(phpstan) : fix phpstan errors in centreon/src (part 1)

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

**Fixes** # MON-11920

Upgrade phpstan to 1.3.0 and php-berbelei to 1.0.0

fixed issues in :
* src/CentreonUser
* src/Security 
* src/kernel.php

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

check if there is errors in
* src/CentreonUser
* src/Security 
* src/kernel.php

### CODE REVIEW

File: src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php [hunk 1]
```diff
@@ -39,6 +39,13 @@ class TimeperiodWebserviceTest extends TestCase
 
     protected const METHOD_GET_LIST = 'getList';
 
+    /**
+     * Undocumented variable
+     *
+     * @var \PHPUnit\Framework\MockObject\MockObject
+     */
```
Comment (jeremyjaouen) [inline src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php:46]:
```suggestion
    /**
     * @var TimeperiodWebservice|\PHPUnit\Framework\MockObject\MockObject
     */
```

File: src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php [hunk 2]
```diff
@@ -62,6 +69,7 @@ protected function setUp(): void
 
     /**
      * Test the method getList
+     * @return void
      */
     public function testGetList()
```
Comment (jeremyjaouen) [inline src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php:74]:
```suggestion
    public function testGetList(): void
```

return void isnt necessary in phpdoc, but should be typehinted in signature

**Review by jeremyjaouen — DISMISSED** [review]

File: src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php (hunk 2 above)
Comment (jeremyjaouen) [reply src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php:74]: i haven't commented the other @return void, i let you handle them ^^

File: src/CentreonUser/Domain/Repository/TimeperiodRepository.php [hunk 3]
```diff
@@ -43,6 +43,7 @@ public static function entityClass(): string
     /**
      * Check list of IDs
      *
+     * @param array<int,int> $ids
```
Comment (adr-mo) [inline src/CentreonUser/Domain/Repository/TimeperiodRepository.php:46]:
```suggestion
     * @param int[] $ids
```

**Review by adr-mo — CHANGES_REQUESTED** [review]

File: src/CentreonUser/Application/Webservice/TimeperiodWebservice.php [hunk 4]
```diff
@@ -33,7 +33,6 @@ class TimeperiodWebservice extends Webservice\WebServiceAbstract implements
     Webservice\WebserviceAutorizeRestApiInterface
 {
     use Webservice\DependenciesTrait;
-
```
Comment (kduret) [inline src/CentreonUser/Application/Webservice/TimeperiodWebservice.php:36]: Do not remove carriage return

File: src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php (hunk 2 above)
Comment (AssanDiomande) [reply src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php]: Done

File: src/CentreonUser/Application/Webservice/TimeperiodWebservice.php (hunk 4 above)
Comment (AssanDiomande) [reply src/CentreonUser/Application/Webservice/TimeperiodWebservice.php]: done

**Review by adr-mo — DISMISSED** [review]

**Review by kduret — CHANGES_REQUESTED** [review]
You need to upgrade phpstan in this PR
Otherwise, we'll not be able to pass CI thresholds

**Comment by AssanDiomande** [comment]
> You need to upgrade phpstan in this PR Otherwise, we'll not be able to pass CI thresholds

If I upgrade the version of phpstan and phpstan-beberlei, the version of other packages (ex monolog) is downgraded (maybe some dependances issues). So I only updated phpstan locally and avoid to push composer files.

**Comment by kduret** [comment]
> > You need to upgrade phpstan in this PR Otherwise, we'll not be able to pass CI thresholds
> 
> If I upgrade the version of phpstan and phpstan-beberlei, the version of other packages (ex monolog) is downgraded (maybe some dependances issues). So I only updated phpstan locally and avoid to push updated composer files.

I don't find any references of monolog dependencies in phpstan & phpstan beberlei repos.
Could you display result of composer command ?

**Comment by AssanDiomande** [comment]
> > > You need to upgrade phpstan in this PR Otherwise, we'll not be able to pass CI thresholds
> > 
> > 
> > If I upgrade the version of phpstan and phpstan-beberlei, the version of other packages (ex monolog) is downgraded (maybe some dependances issues). So I only updated phpstan locally and avoid to push updated composer files.
> 
> I don't find any references of monolog dependencies in phpstan & phpstan beberlei repos. Could you display result of composer command ?

Here what I got : 

Capture

**Comment by kduret** [comment]
I can upgrade phpstan without any downgrade with : 
```shell
composer install
composer require --dev centreon/centreon-test-lib:dev-master
composer require --dev phpstan/phpstan:^1.2.0 phpstan/phpstan-beberlei-assert:^1.0.0
```

**Comment by AssanDiomande** [comment]
> ```shell
> composer require --dev phpstan/phpstan:^1.2.0 phpstan/phpstan-beberlei-assert:^1.0.0
> ```

It's works 👍 
I'll create a specific PR for phpstan update, thanks

File: src/Centreon/Application/DataRepresenter/Topology/NavigationList.php [hunk 5]
```diff
@@ -224,10 +224,10 @@ private function generateLevels($entities, $groups)
     /**
      * Extract the array without keys to avoid serialization into objects
      *
-     * @param  array<array> $naviList
-     * @return array<array>
+     * @param  array<mixed> $naviList
+     * @return array<mixed>
      */
-    private function removeKeysFromArray(array $naviList): array
+    private function removeKeysFromArray(array $naviList)
```
Comment (kduret) [inline src/Centreon/Application/DataRepresenter/Topology/NavigationList.php:230]:
```suggestion
    private function removeKeysFromArray(array $naviList): array
```

File: src/Centreon/Infrastructure/Acknowledgement/AcknowledgementRepositoryRDB.php [hunk 6]
```diff
@@ -124,13 +124,22 @@ public function filterByAccessGroups(?array $accessGroups): AcknowledgementRepos
 
     /**
      * @inheritDoc
+     * @param bool $isAdmin
```
Comment (kduret) [inline src/Centreon/Infrastructure/Acknowledgement/AcknowledgementRepositoryRDB.php:127]: this should be in the interface

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php [hunk 7]
```diff
@@ -588,7 +590,7 @@ public function findHostNamesAlreadyUsed(array $namesToCheck): array
             $names[] = (string) $name;
         }
 
-        if (empty($names)) {
+        if (count($names) == 0) {
```
Comment (kduret) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php:593]: what is the phpstan error here ?

Comment (AssanDiomande) [reply src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php:593]: Variable $names in empty() always exists and is not falsy.

Comment (AssanDiomande) [reply src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php:593]: If i use strict comparison, I get this error : Strict comparison using === between int<1, max> and 0 will always evaluate to false.

Comment (kduret) [reply src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php]: then you can remove `if` statement, cause it cannot be empty

File: src/Centreon/Domain/Monitoring/MonitoringService.php [hunk 8]
```diff
@@ -151,14 +143,14 @@ public function findHostGroups(bool $withHosts = false, bool $withServices = fal
                     $hostGroupIds[] = $hostGroup->getId();
                 }
 
-                if (!empty($hostGroupIds)) {
+                if (count($hostGroupIds) !== 0) {
```
Comment (kduret) [inline src/Centreon/Domain/Monitoring/MonitoringService.php:146]: you can remove this `if` statement

File: src/Centreon/Domain/Monitoring/MonitoringService.php [hunk 9]
```diff
@@ -151,14 +143,14 @@ public function findHostGroups(bool $withHosts = false, bool $withServices = fal
...
                     $hostsByHostsGroups = $this->monitoringRepository->findHostsByHostsGroups($hostGroupIds);
 
                     foreach ($hostGroups as $hostGroup) {
                         if (array_key_exists($hostGroup->getId(), $hostsByHostsGroups)) {
                             $hostGroup->setHosts($hostsByHostsGroups[$hostGroup->getId()]);
                             // We keep the host ids if we must to retrieve their services
-                            if ($withServices && !empty($hostGroup->getHosts())) {
+                            if ($withServices && $hostGroup->getHosts() !== null) {
```
Comment (kduret) [inline src/Centreon/Domain/Monitoring/MonitoringService.php:153]:
```suggestion
                            if ($withServices) {
```

File: src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php [hunk 10]
```diff
@@ -39,6 +39,13 @@ class TimeperiodWebserviceTest extends TestCase
 
     protected const METHOD_GET_LIST = 'getList';
 
+    /**
+     * Undocumented variable
```
Comment (kduret) [inline src/CentreonUser/Tests/Application/Webservice/TimeperiodWebserviceTest.php:43]: then document it :)

File: src/Centreon/Application/DataRepresenter/Topology/NavigationList.php (hunk 5 above)
Comment (kduret) [reply src/Centreon/Application/DataRepresenter/Topology/NavigationList.php]: why did you resolve this conversation ??

File: src/Centreon/Domain/Monitoring/MonitoringService.php [hunk 11]
```diff
@@ -303,11 +295,11 @@ public function findServiceGroupsByHostAndService(int $hostId, int $serviceId):
...
      * Completes hosts with their services.
      *
-     * @param array $hosts Host list for which we want to complete with their services
-     * @return array Returns the host list with their services
+     * @param array<mixed> $hosts Host list for which we want to complete with their services
+     * @return array<mixed> Returns the host list with their services
      * @throws \Exception
      */
-    private function completeHostsWithTheirServices(array $hosts): array
+    private function completeHostsWithTheirServices(array $hosts)
```
Comment (kduret) [inline src/Centreon/Domain/Monitoring/MonitoringService.php:302]:
```suggestion
    private function completeHostsWithTheirServices(array $hosts): array
```

File: src/Centreon/Domain/Entity/Task.php [hunk 12]
```diff
@@ -73,6 +73,14 @@ public function getId(): int
         return $this->id;
     }
 
+    /**
+     * @param int $id
+     */
+    public function setId($id): void
```
Comment (kduret) [inline src/Centreon/Domain/Entity/Task.php:79]:
```suggestion
    public function setId(int $id): void
```

File: src/Centreon/Domain/Monitoring/MonitoringService.php (hunk 9 above)
Comment (AssanDiomande) [reply src/Centreon/Domain/Monitoring/MonitoringService.php:153]: done

File: src/Centreon/Domain/Monitoring/MonitoringService.php (hunk 8 above)
Comment (AssanDiomande) [reply src/Centreon/Domain/Monitoring/MonitoringService.php:146]: done

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php (hunk 7 above)
Comment (AssanDiomande) [reply src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php]: done

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**